examples/server/server.cpp

```diff
@@ -1610,10 +1642,8 @@ struct server_response {
     server_task_result_ptr recv_with_timeout(const std::unordered_set<int> & id_tasks, int timeout) {
         while (true) {
             std::unique_lock<std::mutex> lock(mutex_results);
-            bool cr_res = condition_results.wait_for(lock, std::chrono::seconds(timeout), [&]{
-                return !queue_results.empty();
-            });
-            if (!cr_res) {
+            std::cv_status cr_res = condition_results.wait_for(lock, std::chrono::seconds(timeout));
+            if (cr_res == std::cv_status::timeout) {
```

Comment:
The problem is that if the result comes in at the moment when we're being timeout, this loop will again wait for another result, skipping the result already in the `queue_results`

The simple fix is to move this check to the bottom, so we will read the result queue before `wait_for`
- If we get a result, return without `wait_for`
- If we don't get any result, `wait_for`
  - If `wait_for` timeout, we return nullptr so that the caller can check for HTTP connection state
  - If NOT timeout, we continue to the next iteration of `while (true)`, which again check for incoming result

Reply:
Ok, I'm testing the new version now. The previous one was definitely locking.

Reply:
@ggerganov Were you able to test this? Feel free to let me know if you need help btw

Reply:
Yup, been running without any issues for a while. It's good to merge.